plugin-build/src/main/kotlin/io/sentry/android/gradle/telemetry/SentryTelemetryService.kt

```diff
@@ -0,0 +1,474 @@
+
+            if (isExecAvailable()) {
+                return paramsWithExecAvailable(
+                    project,
+                    cliExecutable,
+                    extension,
+                    variant,
+                    isSaas,
+                    defaultSentryOrganization,
+                    sentryOrg,
+                    buildType,
+                    tags
+                )
+            } else {
+                return SentryTelemetryServiceParams(
+                    extension.telemetry.get(),
+                    extension.telemetryDsn.get(),
+                    sentryOrg,
+                    buildType,
+                    tags,
+                    cliVersion = BuildConfig.CliVersion
+                )
+            }
+        }
+
+        private fun paramsWithExecAvailable(
+            project: Project,
+            cliExecutable: String,
+            extension: SentryPluginExtension,
+            variant: SentryVariant?,
+            isSaas: Boolean?,
+            defaultSentryOrganization: String?,
+            sentryOrg: String?,
+            buildType: String,
+            tags: Map<String, String>
+        ): SentryTelemetryServiceParams {
+            var isSaas1 = isSaas
+            var defaultSentryOrganization1 = defaultSentryOrganization
+            val infoOutput = project.providers.of(SentryCliInfoValueSource::class.java) { cliVS ->
+                cliVS.parameters.cliExecutable.set(cliExecutable)
+                cliVS.parameters.authToken.set(extension.authToken)
+                cliVS.parameters.url.set(extension.url)
+                variant?.let { v ->
+                    cliVS.parameters.propertiesFilePath.set(
+                        SentryPropertiesFileProvider.getPropertiesFilePath(project, v)
+                    )
+                }
+            }.get()
+            isSaas1 = infoOutput.contains("(?m)Sentry Server: .*sentry.io$".toRegex())
+
+            orgRegex.find(infoOutput)?.let { matchResult ->
+                val groupValues = matchResult.groupValues
+                if (groupValues.size > 1) {
+                    val defaultOrg = groupValues[1]
+                    defaultSentryOrganization1 = defaultOrg
+                }
+            }
+
+            return SentryTelemetryServiceParams(
+                extension.telemetry.get(),
+                extension.telemetryDsn.get(),
+                sentryOrg,
+                buildType,
+                tags,
+                cliVersion = BuildConfig.CliVersion
+            )
+        }
+
+        fun register(project: Project): Provider<SentryTelemetryService> {
+            return project.gradle.sharedServices.registerIfAbsent(
+                getBuildServiceName(SentryTelemetryService::class.java),
+                SentryTelemetryService::class.java
+            ) {}
+        }
+
+        private fun isExecAvailable(): Boolean {
+            return GradleVersions.CURRENT >= GradleVersions.VERSION_7_5
```

Comment:
The `ValueSource` class is actually available from Gradle 6.1, so we probably don't need this check at all? https://docs.gradle.org/current/javadoc/org/gradle/api/provider/ValueSource.html

Reply:
Hmm can't remember exactly but I think there was some problem with CI matrix tests where this failed without the check.

Reply:
ok, that's also fine as it is now